File: src/test/java/eu/execom/fabut/model/TierTwoType.java
package eu.execom.fabut.model;

/**
 * Class representing tier two objects with one complex object property.
 * 
 * @author Dusko Vesin
 * @author Nikola Olah
 * @author Bojan Babic
 * @author Nikola Trkulja
 */
public class TierTwoType extends Type {

    /** The Constant PROPERTY. */
    public static final String PROPERTY = "property";

    /** The property. */
    private final TierOneType property;

    /**
     * Instantiates a new tier two type.
     * 
     * @param property
     *            the property
     */
    public TierTwoType(final TierOneType property) {
        this.property = property;
    }

    /**
     * Gets the property.
     * 
     * @return the property
     */
    public TierOneType getProperty() {
        return property;
    }

}

File: src/test/scala/eu/execom/fabut/model/TierTwoType.scala
package eu.execom.fabut.model

/**
 * Class representing tier two objects with one complex object property.
 */
case class TierTwoType(private var _property: TierOneType) extends Type {

  def property: TierOneType = _property

  def property_=(property: TierOneType): Unit = {
    _property = property
  }

  def this() = this(null)
}
